server/src/main/java/org/elasticsearch/index/mapper/ScriptRuntimeField.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+
+public abstract class ScriptRuntimeField implements RuntimeField {
+
+    protected final String name;
```

Comment:
this one corresponds to the simple field name in ObjectMapper right? It is only the name of the leaf field, although it can still contain dots? I wonder if we should clearly document this.

Reply:
We don't have object fields yet so there isn't a distinction here at the moment.  Let's document it when we know precisely how the two different names will work?

Reply:
good point, thanks